offchain/sign_ghost_payment_request.js

```diff
@@ -0,0 +1,226 @@
+
+          return cbk(null, {private_key: privateKey, public_key: unit8AsHex(publicKey)});
+        }
+      ],
+
+      // Assemble the hop hints from the chosen hint channels
+      hints: ['getKeyPair', 'validate', ({getKeyPair}, cbk) => {
+          const routes =  [
+            {
+              public_key: args.destination
+            },
+            {
+              base_fee_mtokens: defaultBaseFee,
+              channel: fakeChannelId,
+              cltv_delta: defaultCltvDelta,
+              fee_rate: defaultFeeRate,
+              public_key: getKeyPair.public_key,
+            },
+          ];
+
+        return cbk(null, [routes]);
+      }],
+
+      // Create the unsigned payment request
+      unsigned: ['hints', 'getKeyPair', ({getKeyPair, hints}, cbk) => {
+        try {
+          const unsigned = createUnsignedRequest({
+            cltv_delta: args.cltv_delta,
+            description: args.description,
+            destination: getKeyPair.public_key,
+            features: args.features,
+            id: args.id,
+            network: args.network,
+            payment: args.payment,
+            routes: !!hints.length ? hints : undefined,
+            tokens: args.tokens,
+          });
+
+          return cbk(null, unsigned);
+        } catch (err) {
+          return cbk([500, 'UnexpectedErrorCreatingUnsignedRequest', {err}]);
+        }
+      }],
+
+      // Sign the unsigned payment request
+      sign: ['getKeyPair', 'unsigned', ({getKeyPair, unsigned}, cbk) => {
+        const signature = tinysecp256k1.sign(hexAsBuffer(unsigned.hash), getKeyPair.private_key);
+
+        return cbk(null, {signature: unit8AsHex(signature)});
+      }],
+
+      // Assemble the signed request
+      request: [
+        'getKeyPair', 
+        'sign', 
+        'unsigned', 
+        ({getKeyPair, sign, unsigned}, cbk) => {
+        try {
+          const {request} = createSignedRequest({
+            destination: getKeyPair.public_key,
+            hrp: unsigned.hrp,
+            signature: sign.signature,
+            tags: unsigned.tags,
+          });
+
+          args.logger.info({
+            request,
+            tokens: args.tokens
+          });
+          
+          return cbk(null, {request, tokens: args.tokens});
+        } catch (err) {
+          return cbk([503, 'UnexpectedErrorSigningRequest', {err}]);
+        }
+      }],
+
+      // Subscribe and intercept forward request
+      intercept: [
+        'request',
+        'unsigned',
+        ({}, cbk) => {
+          args.logger.info({
+            is_intercepting_forward: true,
+          });
+
+          const sub = subscribeToForwardRequests({lnd: args.lnd});
+
+          const finished = (err, res) => {
+            sub.removeAllListeners();
+
+            return cbk(err, res);
+          };
+          
+          sub.on('forward_request', forward => {
+            if (args.id === forward.hash) {
+              forward.settle({secret: args.secret})
```

Comment:
shouldn't it exit here

Reply:
There seems to be an issue where it's exiting too early after settling and the payment is failing so I didn't exit. Didn't know how to deal with it.

Reply:
interesting, maybe there is a problem in ln-service